src/main/scala/com/microsoft/hyperspace/index/plans/logical/IndexHadoopFsRelation.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.plans.logical
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.catalog.BucketSpec
+import org.apache.spark.sql.execution.datasources.{FileFormat, FileIndex, HadoopFsRelation}
+import org.apache.spark.sql.types.StructType
+
+import com.microsoft.hyperspace.index.IndexLogEntry
+
+/**
+ * Wrapper class of HadoopFsRelation to indicate index application more explicitly in Plan string.
+ */
+class IndexHadoopFsRelation(
```

Comment:
is there a reason for choosing class over case class, same as HadoopFsRelation (since it's a wrapper, I would think both are of same type)

Reply:
Because case-to-case inheritance is not proper in Scala.